packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts

```diff
@@ -0,0 +1,333 @@
+import {
+    FormState,
+    PrecomposedTransactionFinal,
+    ComposeActionContext,
+    ExternalOutput,
+    PrecomposedTransaction,
+    PrecomposedLevels,
+} from '@suite-common/wallet-types';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import { selectBlockchainBlockHashBySymbol, selectDevice } from '@suite-common/wallet-core';
+import { Dispatch, GetState } from 'src/types/suite';
+import {
+    amountToSatoshi,
+    calculateMax,
+    calculateTotal,
+    formatAmount,
+    getExternalComposeOutput,
+} from '@suite-common/wallet-utils';
+import BigNumber from 'bignumber.js';
+import {
+    getPubKeyFromAddress,
+    buildTransferTransaction,
+    buildTokenTransferTransaction,
+} from 'src/utils/wallet/solanaUtils';
+import { SYSTEM_PROGRAM_PUBLIC_KEY } from '@trezor/blockchain-link-utils/lib/solana';
+
+const calculate = (
+    availableBalance: string,
+    output: ExternalOutput,
+    feeLevel: FeeLevel,
+    token?: TokenInfo,
+    destinationAddress?: string,
+    isCreatingAccount?: boolean,
+): PrecomposedTransaction => {
+    const feeInLamports = feeLevel.feePerUnit;
+    let amount: string;
+    let max: string | undefined;
+    const availableTokenBalance = token
+        ? amountToSatoshi(token.balance!, token.decimals)
+        : undefined;
+    if (output.type === 'send-max' || output.type === 'send-max-noaddress') {
+        max = availableTokenBalance || calculateMax(availableBalance, feeInLamports);
+        amount = max;
+    } else {
+        amount = output.amount;
+    }
+
+    // total SOL spent (amount + fee), in case of SPL token only the fee
+    const totalSpent = new BigNumber(calculateTotal(token ? '0' : amount, feeInLamports));
+
+    if (totalSpent.isGreaterThan(availableBalance)) {
+        const error = token ? 'AMOUNT_NOT_ENOUGH_CURRENCY_FEE' : 'AMOUNT_IS_NOT_ENOUGH';
+        // errorMessage declared later
+        return { type: 'error', error, errorMessage: { id: error } } as const;
+    }
+
+    const payloadData: PrecomposedTransaction = {
+        type: 'nonfinal',
+        totalSpent: token ? amount : totalSpent.toString(),
+        max,
+        fee: feeInLamports,
+        feePerByte: feeInLamports,
+        token,
+        bytes: 0,
+        inputs: [],
+    };
+
+    if (output.type === 'send-max' || output.type === 'payment') {
+        return {
+            ...payloadData,
+            type: 'final',
+            // compatibility with BTC PrecomposedTransaction from @trezor/connect
+            inputs: [],
+            outputsPermutation: [0],
+            outputs: [
+                {
+                    // TODO: internationalise this!
+                    address:
+                        destinationAddress && destinationAddress !== output.address
+                            ? `${destinationAddress} (Token account for: ${output.address} ${
+                                  isCreatingAccount && ', Account will be created.'
```

Comment:
you can't use `boolean &&` like this, otherwise "false" is rendered, see 
<img width="408" alt="image" src="https://github.com/trezor/trezor-suite/assets/3729633/d1e2521e-17e5-4beb-9b1d-1201260a26c2">

```suggestion
                            ? `${destinationAddress} (Token account for: ${output.address}${
                                  isCreatingAccount ? ' , Account will be created.' : ''
```

Reply:
Oops, react brain got to me.